tests/e2e/specs/miscellaneous/CustomOpenVSXRegistry.spec.ts

```diff
@@ -0,0 +1,217 @@
+			expect(await projectAndFileTests.getProjectTreeItem(projectSection, testRepoProjectName), 'Project folder was not imported').not
+				.undefined;
+		});
+		test('Accept the project as a trusted one', async function (): Promise<void> {
+			await projectAndFileTests.performTrustAuthorDialog();
+		});
+		test('Edit VSX plugin list', function (): void {
+			// modify the openvsx-sync.json file with just one item - for speeding up the build an image
+			kubernetesCommandLineToolsExecutor = e2eContainer.get(CLASSES.KubernetesCommandLineToolsExecutor);
+			kubernetesCommandLineToolsExecutor.workspaceName = workspaceName;
+			kubernetesCommandLineToolsExecutor.loginToOcp('admin');
+			kubernetesCommandLineToolsExecutor.getPodAndContainerNames();
+			const pathToOpenVSXOriginFile: string = `${pathToPluginRegistry}openvsx-sync.json`;
+			const pathToOpenVSXTmpFile: string = `${pathToPluginRegistry}temp.json`;
+			const editVSCListCommand: string = `jq "[.[] | select(.id == \\"redhat.vscode-xml\\")]" ${pathToOpenVSXOriginFile} > ${pathToOpenVSXTmpFile} && mv ${pathToOpenVSXTmpFile} ${pathToOpenVSXOriginFile}`;
+			const output: ShellString = kubernetesCommandLineToolsExecutor.execInContainerCommand(editVSCListCommand);
+			expect(output.code).to.equals(0);
+			const getCheClusterNamespaceCommand: string = 'oc get checluster --all-namespaces -o json | jq -r ".items[0].metadata.name"';
+			const getCheClusterNameSpace: string = 'oc get checluster --all-namespaces -o json | jq -r ".items[0].metadata.namespace"';
+
+			// get che cluster name and namespace and init variables
+			cheClusterName = kubernetesCommandLineToolsExecutor.execInContainerCommand(getCheClusterNamespaceCommand).replace('\n', '');
+			cheClusterNamespace = kubernetesCommandLineToolsExecutor.execInContainerCommand(getCheClusterNameSpace).replace('\n', '');
+		});
+
+		test('Login podman into official registry', function (): void {
+			const commandToAuthenticateInRegistry: string = `podman login -u \"${process.env.REGISTRY_LOGIN}\" -p ${process.env.REGISTRY_PASSWORD} registry.redhat.io`;
+			const output: ShellString = kubernetesCommandLineToolsExecutor.execInContainerCommand(commandToAuthenticateInRegistry);
+			expect(output).contains('Login Succeeded!');
+		});
+
+		test('Build and push custom image', function (): void {
+			currentNamespace = kubernetesCommandLineToolsExecutor.namespace;
+			const podmanPushCommand: string = `podman push ${internalRegistry}/${currentNamespace}/che-plugin-registry:${imageTag}`;
+			const commandToBuildCustomVSXImage: string = `cd ${pathToPluginRegistry} && yes | ./build.sh`;
+			const commandLoginIntoInternalRegistry: string =
+				'podman login -u $(oc whoami | tr -d :) -p $(oc whoami -t) image-registry.openshift-image-registry.svc:5000';
+			const retagImageCommand: string = `podman tag quay.io/devspaces/pluginregistry-rhel8:next  ${internalRegistry}/${currentNamespace}/che-plugin-registry:${imageTag}`;
```

Comment:
From Dev Spaces pre-testing perspective it needs to be "pluginregistry-rhel8:**latest**".
You could probably re-use `TESTING_APPLICATION_VERSION` if ignore remove bug fix version.

Reply:
Add logic here: https://github.com/eclipse-che/che/pull/23092/commits/07cbb1bc9b06f4d9fcb8f736c6149a1779e6a0f4#diff-32cc0512c9bbd2a2597fa64ee14044dbec9d68a78077bea8a62c80e28fb26d0fR42

Reply:
thank you